### DISCUSSION TITLE
Create subwindows with winit

### DISCUSSION THREAD

**Discussion by J-Cake** [vizia/vizia#388] [Questions]
Hi folks, I'd like to add a separate region to my application using a subwindow. I'd like to avoid the `Popup` for multiple reasons, such as the complexity of the content I'd like to display, and to allow the user to handle the window using the system.

Without having to reimplement the `Application` struct, what might my options be?

**Comment by geom3trik** [comment]
Unfortunately support for multiple windows is still work in progress so this is not yet possible.

**Reply by J-Cake** [reply]
Are there any specific areas of development, where I could potentially make myself useful?

**Reply by geom3trik** [reply]
My latest effort for multi-window can be found on this branch: <https://github.com/vizia/vizia/tree/multiwindow> (though it needs rebasing). Spawning of multiple windows works fine but the tricky part I have yet to figure out is sharing OpenGL resources, such as the textures used for text atlases, between the windows. Part of this involves setting up a shared context when creating the windows <https://docs.rs/glutin/latest/glutin/context/struct.ContextAttributesBuilder.html#method.with_sharing>, and the other part is using native textures to build the font atlas <https://docs.rs/femtovg/latest/femtovg/struct.Canvas.html#method.create_image_from_native_texture>. Unfortunately my OpenGL knowledge is limited so I haven't gotten back to this attempt yet.

**Reply by J-Cake** [reply]
Ah I see. Unfortunately, I can say that I have even less expertise with OpenGL, but I will def take a look. If I find anything, I'll PR etc... 

Thanks for your help though

**Reply by geom3trik** [reply]
Another easier option would be to have a separate context per window. This would mean each window would have its own font atlas, which would mean some duplication but might not be so bad for a first attempt. The same problem would apply to images as well though, they would have to be loaded per window. Maybe not a problem if your images are only used in one window or another.

**Reply by J-Cake** [reply]
Actually, my app at the moment is mostly heavy on text, so fonts, but it's not rich text, so a single font would actually be sufficient. There are no images, so in my particular use-case, that's fine.

Of course, true multiwindow would be amazing
